picard/formats/__init__.py

```diff
@@ -39,17 +40,47 @@ def supported_formats():
     return formats
 
 
+def guess_format(filename, options=_formats):
+    """Select the best matching file type amongst supported formats."""
+    results = []
+    with file(filename, "rb") as fileobj:
```

Comment:
> We already have an open() method defined in the file. Which is why I opted to use file()

Yes, and it is perhaps not the best thing. This module level `open()` isn't great at all vs the builtin `open()`.

About `file()`, it is removed from Python 3.0, with comment "Use open" (https://docs.python.org/release/3.0/whatsnew/3.0.html#builtins)